plugins/poll/assets/javascripts/discourse/components/poll-results-standard.gjs

```diff
@@ -0,0 +1,88 @@
+import Component from "@glimmer/component";
+import { concat } from "@ember/helper";
+import { htmlSafe } from "@ember/template";
+import evenRound from "discourse/plugins/poll/lib/even-round";
+import PollVoters from "./poll-voters";
+
+export default class PollResultsStandardComponent extends Component {
+  get votersCount() {
+    return this.args.votersCount || 0;
+  }
+
+  get orderedOptions() {
+    const votersCount = this.votersCount;
+
+    let ordered = [...this.args.options].sort((a, b) => {
+      if (a.votes < b.votes) {
+        return 1;
+      } else if (a.votes === b.votes) {
+        if (a.html < b.html) {
+          return -1;
+        } else {
+          return 1;
+        }
+      } else {
+        return -1;
+      }
+    });
+
+    const percentages =
+      votersCount === 0
+        ? Array(ordered.length).fill(0)
+        : ordered.map((o) => (100 * o.votes) / votersCount);
+
+    const rounded = this.isMultiple
+      ? percentages.map(Math.floor)
+      : evenRound(percentages);
+
+    ordered.forEach((option, idx) => {
+      const per = rounded[idx].toString();
+      const chosen = (this.args.vote || []).includes(option.id);
+      option.percentage = per;
+      option.chosen = chosen;
+      let voters = this.args.voters[option.id] || [];
+      option.voters = [...voters];
+    });
+
+    return ordered;
+  }
```

Comment:
method is too complex

Reply:
this is mostly a lift and shift from the widget code: https://github.com/discourse/discourse/blob/581dbca97f2b55c9bbbe40dc3b58a9df7409d77f/plugins/poll/assets/javascripts/discourse/widgets/discourse-poll.js#L173

Are you concerned that it will be too slow in Glimmer?

Reply:
@jjaffeux made many improvements to this in response to your feedback:

* https://github.com/discourse/discourse/pull/27155/commits/531160cd59f33e6ac6b7ec2c325bb738af30888f
* https://github.com/discourse/discourse/pull/27155/commits/7bb4c5d4a5bbde4a82de74b0c647a1ed83c48ba2
* https://github.com/discourse/discourse/pull/27155/commits/c0a763c1a080c90dd767f82c22973ccbcdef19c8